challenge_14/c/karanchawla/challenge_14.c

```diff
@@ -0,0 +1,90 @@
+//Karan Chawla
+//Challenge 14
+
+#include <stdio.h>
+#include <stdlib.h>
+
+//Linked list node definition
+typedef struct node
+{
+	char c;
+	struct node* next;
+}Node;
+
+
+//utility function to create a new node
+Node* newNode(char c)
+{
+	Node* NewNode = (Node*)malloc(sizeof(Node));
+
+	NewNode->c = c;
+	NewNode->next = NULL;
+
+	return NewNode;
+}
+
+//utility function to add a new node to the linked list
+void push(Node** head_ref, char c)
+{
+	Node* new_node = newNode(c);
+
+	new_node->next = (*head_ref);
+
+	(*head_ref) = new_node;
+}
+
+
+//function to reverse the linked list
+//prev->curr->next
+//changes prev to next and moves current along the linked list until NULL
+void reverseLinkedList(Node** head_ref)
+{
+	if (*head_ref==NULL)
+		return;
+
+	Node* prev = NULL;
+	Node* current = *head_ref;
+	Node* next;
+	while(current!=NULL)
+	{
+		next = current->next;
+		current->next = prev;
+		prev = current;
+		current = next;
+	}
+	(*head_ref) = prev;
+}
+
+//utility function to print linked list
+void printLinkedList(Node* head)
+{
+	while(head!=NULL)
+	{
+		printf("%c->", head->c);
+		head = head->next;
+	}
+	printf("NULL");
```

Comment:
Made the changes. And yes shouldn't wrap my de-refs in parenthesis. Thanks for the review :)